charts/pipelines/templates/tasks/static-code-analysis.yaml

```diff
@@ -0,0 +1,29 @@
+{{ if eq .Values.static_code_analysis true }}
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: static-code-analysis
+spec:
+  workspaces:
+    - name: output
+    - name: sonarqube-auth
+      optional: true
+  params:
+    - name: WORK_DIRECTORY
+      description: Directory to start build in (handle multiple branches)
+      type: string
+    - name: APPLICATION_NAME
+      description: Name of the application
+      type: string
+  steps:
+  - name: sonar-scan
+    workingDir: $(workspaces.output.path)/$(params.WORK_DIRECTORY)
+    image: quay.io/rlundber/mlops-ct-cicd:0.4
+    command: ["/bin/sh", "-c"]
+    args:
+    - |
+      python3 -m pip pysonar-scanner
+      pysonar-scanner -Dsonar.host.url=http://sonarqube-sonarqube:9000 -Dsonar.projectKey=$(params.APPLICATION_NAME) -Dsonar.login=$(cat $(workspaces.sonarqube-auth.path)/username) -Dsonar.password=$(cat $(workspaces.sonarqube-auth.path)/password)
```

Comment:
@ckavili where is sonarqube deployed? will be in lab-deploy?

Reply:
they will be deploy in their own mlops namespace where the pipeline runs 🤸🏻